perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -555,6 +574,20 @@ public void moveGroup(PerunSession sess, Group destinationGroup, Group movingGro
 		// And finally update parentGroupId for moving group in database
 		this.updateParentGroupId(sess, movingGroup);
 
+		List<Member> movingGroupMembers = getGroupMembers(sess, movingGroup);
+
+		for(Member member : movingGroupMembers) {
+
+			// calculate new member-group statuses for members in moving group with new parent group
+			recalculateMemberGroupStatusRecursively(sess, member, movingGroup);
+
+			if (previousParent != null) {
+
+				// calculate new member-group statuses for members from previous moving group parent
+				recalculateMemberGroupStatusRecursively(sess, member, previousParent);
+			}
+		}
```

Comment:
I am not sure if this is correct. Recalculation should be part of adding and removing membership in groups. Try to answer situation like this:
1] G1 is top level group.
2] G2 is subgroup of group G1.
3] G3 is subgroup of group G2.
4] M1 is member and it is directly in G1 and G3. It is indirectly in G2 from G3.
5] In group G1, member M1 has status EXPIRED and in G3 he has status VALID. So final status in G1 is VALID beacuse of status in G3.
6] STEP: I will move G3 to the TOP. So member M1 will stay in G3 direct and VALID. And also M1 will stay in G1 direct and VALID. But will be removed from G2 at all.
7] Now you are calling "recalculcateMemberGroupStatusRecursively" on G3 and also on G2. G3 will proceed successfully, but what with G2? How can proceed when there is no more member M1 in this group?

Will this code work on this scenario correctly?

Reply:
Yes, it will work. I have created another test to test this specific scenario. The test name is 'moveGroupCorrectMemberGroupStatuses2' and is in groupEntryIntegrationTests. There is also one more complex test that tests this functionality.